api.ts

```diff
@@ -372,6 +372,14 @@ export type ParameterOptions<T extends ParameterType> = Omit<ParamDef<ParameterT
     : undefined;
 };
 
+type ParamDefFromOptionsUnion<T extends ParameterType, O extends ParameterOptions<T>> = Omit<
+  O,
+  'type' | 'autcomplete'
+> & {
+  type: O extends ParameterOptions<infer S> ? ParameterTypeMap[S] : never;
```

Comment:
i honestly have no idea why this works...

maybe because [`OptionalParamDef<infer S>`](https://github.com/coda/packs-sdk/blob/4f3b641575b031e1620e1951b75db48e7ee6ecd5/api_types.ts#L346) is also using inferring S?

Reply:
![image](https://user-images.githubusercontent.com/70549840/191899078-5685a75b-ff53-43cb-a4a7-3a70e08992fe.png)
![image](https://user-images.githubusercontent.com/70549840/191899090-384c0ca9-5d44-4c87-8ede-00970a5f09b6.png)